sandbox/apparmor/apparmor.go

```diff
@@ -56,6 +57,221 @@ func ValidateNoAppArmorRegexp(s string) error {
 	return nil
 }
 
+type AAREExclusionPatternsOptions struct {
+	// Prefix is a string to include on every line in the permutations before
+	// the exclusion permutation itself.
+	Prefix string
+	// Suffix is a string to include on every line in the permutations after
+	// the exclusion permutation itself.
+	Suffix string
+
+	// TODO: add options for generating non-filepaths like what we need for the
+	// unconfined profile transition exclusion in snap-confine's profile as well
+	// as an option for adding an extra character to the very first rule like we
+	// need in the home interface
+}
+
+func InsertAAREExclusionPatterns(aaRules string, excludePatterns []string, opts *AAREExclusionPatternsOptions) (string, error) {
+
```

Comment:
Stray newline

Reply:
Done in commit 25f75be5889fc9b06f51856734f4fb712dfe42fa